cloud-service-provider/azure/aks/terraform/azure_main.tf

```diff
@@ -21,6 +21,70 @@ module "vnet" {
   depends_on = [azurerm_resource_group.main]
 }
 
+# Cosmos DB
+resource "azurerm_cosmosdb_account" "main" {
+  name                      = "${var.cluster_name}-cosmosdb"
+  location                  = var.cosmosdb_account_location
+  resource_group_name       = azurerm_resource_group.main.name
+  offer_type                = "Standard"
+  kind                      = "GlobalDocumentDB"
+  geo_location {
+    location          = var.cosmosdb_account_location
+    failover_priority = 0
+  }
+  consistency_policy {
+    consistency_level       = "BoundedStaleness"
+    max_interval_in_seconds = 300
+    max_staleness_prefix    = 100000
+  }
+  capabilities {
+    name = "EnableNoSQLFullTextSearch"
+  }
+  capabilities {
+    name = "EnableNoSQLVectorSearch"
+  }
+  depends_on = [
+    azurerm_resource_group.main
+  ]
+}
+
+resource "azurerm_cosmosdb_sql_database" "main" {
+  name                = "${var.cluster_name}-sqldb"
+  resource_group_name = azurerm_resource_group.main.name
+  account_name        = azurerm_cosmosdb_account.main.name
+  throughput          = var.throughput
+}
+
+resource "azurerm_cosmosdb_sql_container" "main" {
+  name                  = "${var.cluster_name}-sql-container"
+  resource_group_name   = azurerm_resource_group.main.name
+  account_name          = azurerm_cosmosdb_account.main.name
+  database_name         = azurerm_cosmosdb_sql_database.main.name
+  partition_key_paths   = ["/definition/id"]
+  partition_key_version = 1
+  throughput            = var.throughput
+
+  indexing_policy {
+    indexing_mode = "consistent"
+
+    included_path {
+      path = "/*"
```

Comment:
Does not /* cover everything? Is there some ordering effect here .. from most general to specific?